lib/cmr/response.rb

```diff
@@ -41,24 +41,27 @@ def body_is_html?
     end
 
     def errors
-      @errors ||= if body_is_html?
-                    Rails.logger.error "CMR Error Response Body is a HTML document: #{body}"
-                    ['There was an error with the operation you were trying to perform. There may be an issue with one of the services we depend on. Please contact your provider administrator or the CMR OPS team.']
-                  elsif body.is_a?(Hash)
-                    if body['errors']
-                      Array.wrap(body['errors'])
-                    elsif body['error']
-                      Array.wrap(body['error'])
-                    end
-                  elsif body.is_a?(String)
-                    [body]
-                  else
-                    []
+      return @errors if @errors.present?
```

Comment:
what was wrong with the `||=` ?

Reply:
Nothing was wrong with it, but we can minimize the number of times we search the errors for tokens if we do it only when we assign them here.  
Using the ||= syntax would have meant each part of the if tree would have needed the call to scrub the tokens or the errors would be checked for tokens every time the errors method was called.
Either: 
`@errors ||= if a
                       do thing
                       check for tokens
                     elsif b
                       do thing
                       check for tokens...`
Or: 
`@errors ||= all of this unchanged
check for tokens
@errors` where the check for tokens occurs every time this is called...